app/lib/search/index_simple.dart

```diff
@@ -227,29 +231,70 @@ class SimplePackageIndex implements PackageIndex {
 }
 
 class Score {
-  final Map<String, double> values = <String, double>{};
+  final Map<String, double> _values;
 
-  Iterable<String> getKeys() => values.keys;
+  Score(Map<String, double> values) : _values = new Map.unmodifiable(values);
 
-  void addValues(Map<String, double> newValues, double weight) {
-    if (newValues == null) return;
-    newValues.forEach((String key, double score) {
-      if (score != null) {
-        final double prev = values[key] ?? 0.0;
-        values[key] = prev + score * weight;
-      }
-    });
+  Set<String> getKeys() => _values.keys.toSet();
+  double getMaxValue() => _values.values.fold(0.0, max);
+  Map<String, double> getValues() => _values;
+
+  double operator [](String key) => _values[key] ?? 0.0;
+
+  /// Calculates the intersection of the two Scores, by multiplying the values.
+  ///
+  /// [weight] can be used to dampen the removal-effect of multiplication with
+  /// zero or very low values, and to keep results from the original set.
+  Score multiply(Score other, {double weight: 0.0}) {
```

Comment:
Maybe it would be better making this a static method and rename it to `interpolate`:
```dart
/// Parameter t in [0, 1] interpolates between [scoreA] and [scoreB].
/// The resulting [score]'s keys are the common keys between [scoreA] and [scoreB]
Score.interpolate(scoreA, scoreB, 0.1);
````

Reply:
I've added interpolate, but kept the multiple (rewrote it a bit similar to `max`). The interpolation as direct part of the scoring is not great because it can not only decrease good scores, but also it can increase bad scores, which is not what we want here.